src/hello_freertos.c

```diff
@@ -1,54 +1,213 @@
+
+    while (true) {
+        gpio_put(OUT_PIN, true); // Turn on LED
+        vTaskDelay(1500);         // Wait 1500ms
+        gpio_put(OUT_PIN, false); // Turn off LED
+        vTaskDelay(250);         // Wait 1500ms
+    }
+}
+
+void blink_task_thread2(__unused void *params) {
+    gpio_init(OUT_PIN);            // Initialize the GPIO pin
+    gpio_set_dir(OUT_PIN, GPIO_OUT); // Set it as an output pin
+
+    while (true) {
+        gpio_put(OUT_PIN, true); // Turn on LED
+        vTaskDelay(2000);         // Wait 2000ms
+        gpio_put(OUT_PIN, false); // Turn off LED
+        vTaskDelay(2000);         // Wait 2000ms
+    }
+}
+
+// 3. Run a busy loop to get 100% CPU utilization
+void blink_task_busy_loop(__unused void *params) {
+    gpio_init(OUT_PIN);            // Initialize the GPIO pin
+    gpio_set_dir(OUT_PIN, GPIO_OUT); // Set it as an output pin
+
+    while (true) {
+        // Blink LED rapidly in a busy loop
+        gpio_put(OUT_PIN, true);
+        for (volatile int i = 0; i < 1000000; i++); // Busy delay
+        gpio_put(OUT_PIN, false);
+        for (volatile int i = 0; i < 1000000; i++); // Busy delay
+    }
+}
+
+// 4. Disable compiler optimizations with a tight loop
+#pragma GCC optimize ("O0") // Disable compiler optimizations for this function
+void blink_task_no_optimize(__unused void *params) {
+    gpio_init(OUT_PIN);            // Initialize the GPIO pin
+    gpio_set_dir(OUT_PIN, GPIO_OUT); // Set it as an output pin
+
+    while (true) {
+        gpio_put(OUT_PIN, true);    // Turn on LED
+
+        // Tight loop for CPU usage (no delay)
+        for (int i = 0; i < 10000000; i++) {
+            // Perform some trivial but CPU-heavy operations (e.g., multiplication)
+            volatile int dummy = i * 3;  // Keeps CPU busy with computation
+        }
+
+        gpio_put(OUT_PIN, false);   // Turn off LED
+
+        // Tight loop for CPU usage (no delay)
+        for (int i = 0; i < 10000000; i++) {
+            // Another CPU-heavy operation
+            volatile int dummy = i * 2;  // Keeps CPU busy with computation
+        }
+    }
+}
+
+// 5. Run a tight loop to trigger high CPU activity
+void blink_task_tight_loop(__unused void *params) {
+    gpio_init(OUT_PIN);            // Initialize the GPIO pin
+    gpio_set_dir(OUT_PIN, GPIO_OUT); // Set it as an output pin
+
+    volatile int result = 0;
+    while (true) {
+        for (volatile int i = 1; i <= 10000; i++) {
+            result += i * i; // Perform multiply and accumulate
+        }
+        gpio_put(OUT_PIN, !gpio_get(OUT_PIN)); // Toggle LED
     }
 }
 
-int main( void )
-{
+int main(void) {
+    // Initialize standard I/O for printing messages
     stdio_init_all();
-    const char *rtos_name;
-    rtos_name = "FreeRTOS";
-    TaskHandle_t task;
-    xTaskCreate(main_task, "MainThread",
-                MAIN_TASK_STACK_SIZE, NULL, MAIN_TASK_PRIORITY, &task);
+
+    // Uncomment the desired task to run ---------
```

Comment:
Using conditional compilation with macros and using cmake's `target_compile_definitions` to create separate executables from the same source file is probably a better alternative to doing this.